### PR TITLE
feat: add tx fee checker to `x/subspace` ante

### PR DESCRIPTION
**Pull request by dadamu**
## Description

This PR updates x/subspace fee deduction ante to have tx fee checker as what previously mempool fee decorator does.

Depends-On: #1097 
Forwarding: https://github.com/desmos-labs/desmos/pull/1097#discussion_r1155856592

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [ ] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by dadamu** [comment]
Waiting #1107

File: x/subspaces/ante/fee.go [hunk 1]
```diff
@@ -40,16 +48,30 @@ func (dfd DeductFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bo
 
 	feeTx, ok := tx.(sdk.FeeTx)
 	if !ok {
-		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+		return ctx, errors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+	}
+
+	if !simulate && ctx.BlockHeight() > 0 && feeTx.GetGas() == 0 {
```
Comment (dadamu) [inline x/subspaces/ante/fee.go:54]: In simulation mod, the gas fees can be 0.
In addition, the genesis txs are also 0 gas txs.

File: x/subspaces/ante/fee.go [hunk 2]
```diff
@@ -40,16 +48,30 @@ func (dfd DeductFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bo
...
+		return ctx, sdkerrors.Wrap(sdkerrors.ErrInvalidGasLimit, "must provide positive gas")
+	}
+
+	fees := feeTx.GetFee()
+	var priority int64
+	if !simulate {
```
Comment (dadamu) [inline x/subspaces/ante/fee.go:60]: As above

File: x/subspaces/ante/fee_test.go [hunk 3]
```diff
@@ -194,6 +251,10 @@ func (suite *AnteTestSuite) TestAnte_Ante() {
 				suite.Require().Error(err)
 			} else {
 				suite.Require().NoError(err)
+
```
Comment (RiccardoM) [inline x/subspaces/ante/fee_test.go:254]:
```suggestion
```

**Review by RiccardoM — APPROVED** [review]

**Review by manu0466 — APPROVED** [review]

**Pull request merged**